apps/meteor/ee/app/license/server/lib/isUnderAppLimits.ts

```diff
@@ -1,13 +1,19 @@
-import type { AppManager } from '@rocket.chat/apps-engine/server/AppManager';
+import { Apps } from '@rocket.chat/core-services';
 
 import type { ILicense, LicenseAppSources } from '../../definition/ILicense';
 
-export async function isUnderAppLimits(
-	{ appManager }: { appManager: AppManager },
-	licenseAppsConfig: NonNullable<ILicense['apps']>,
-	source: LicenseAppSources,
-): Promise<boolean> {
-	const apps = appManager.get({ enabled: true }).filter((app) => app.getStorageItem().installationSource === source);
+export async function isUnderAppLimits(licenseAppsConfig: NonNullable<ILicense['apps']>, source: LicenseAppSources): Promise<boolean> {
+	const apps = await Apps.getApp({ enabled: true });
+
+	if (!apps || !Array.isArray(apps)) {
+		return true;
+	}
+
+	const promisedStorageItems = apps.map(async (app) => Apps.getAppStorageItemById(app.id));
+	const activeAppsFromSameSource = await Promise.all(promisedStorageItems).then((items) =>
+		items.filter((item) => item && item.installationSource === source),
+	);
```

Comment:
it feels very confusing if you mix `await` and `then`, specially on the same command.

```suggestion
	const storageItems = await Promise.all(apps.map(async (app) => Apps.getAppStorageItemById(app.id)));
	const activeAppsFromSameSource = storageItems.filter((item) => item && item.installationSource === source);
```